derp/derp_server.go

```diff
@@ -1829,19 +1824,27 @@ func (c *sclient) setWriteDeadline() {
 		// of connected peers.
 		d = privilegedWriteTimeout
 	}
-	c.nc.SetWriteDeadline(time.Now().Add(d))
+	if d <= 0 {
```

Comment:
I don't expect us to use this, but in case someone doesn't configure the `Server` with a timeout, I think it's better to set no timeout than to set a 0 timeout, which would just result in all writes failing.